cmd/symbols/internal/symbols/search.go

```diff
@@ -109,57 +134,212 @@ func filterSymbols(ctx context.Context, symbols []protocol.Symbol, args protocol
 		}
 		span.Finish()
 	}()
-	span.SetTag("before", len(symbols))
 
-	query := args.Query
-	if !args.IsRegExp {
-		query = regexp.QuoteMeta(query)
+	const maxFirst = 500
+	if args.First < 0 || args.First > maxFirst {
+		args.First = maxFirst
 	}
-	if !args.IsCaseSensitive {
-		query = "(?i:" + query + ")"
+
+	makeCondition := func(column string, regex string) []*sqlf.Query {
+		conditions := []*sqlf.Query{}
+
+		if regex == "" {
+			return conditions
+		}
+
+		if strings.HasPrefix(regex, "^") && strings.HasSuffix(regex, "$") {
+			// It looks like the user is asking for exact matches, so use `=`
+			// for speed. Checking for `^...$` isn't 100% accurate, but it
+			// covers 99.9% of cases.
+			symbolName := strings.TrimSuffix(strings.TrimPrefix(regex, "^"), "$")
```

Comment:
Slice would probably be faster because it doesn't need to check again if the string starts/ends with the characters and doesn't need a function call
```suggestion
			symbolName := regex[1:len(regex)-1]
```

Reply:
For each query (e.g. one basic code intel hover), this is called once for the query, once for the file exclude pattern, and once for each file include pattern (and usually there is 1 of those), which makes me think this is not a performance concern. The intent to strip the `^...$` is clearer when written using `Trim*` than with `regex[1:len(regex)-1]`.